src/core/Akka/Util/Internal/TaskExtensions.cs

```diff
@@ -40,5 +40,55 @@ public static Task WithCancellation(this Task task, CancellationToken cancellati
                 // Check cancellation, to return task in cancelled state instead of completed
                 .ContinueWith(t => { }, cancellationToken, TaskContinuationOptions.ExecuteSynchronously, TaskScheduler.Default);
         }
+
+        /// <summary>
+        /// When this Task is completed, either through an exception or a value, invoke the provided function.
+        /// If the Task has already been completed, this will either be applied immediately or be scheduled asynchronously.
+        /// </summary>
+        /// <param name="source">TBD</param>
+        /// <param name="f">The function to be executed when this Task completes</param>
+        public static Task OnComplete(this Task source, Action<Try<Done>> f)
```

Comment:
I understand these extensions are not idiomatic in C#, but working with `ContinueWith` is cumbersome when you have to repeat a chunk of code like this one over and over again.

Maybe porting more of these extensions from the JVM wouldn't be such a bad idea after all.

Reply:
This stuff is all marked as part of our internal APIs anyways - so users shouldn't expect that this would be part of the public-facing API that we preserve. If these make things easier, we should include them.